### PR TITLE
Introduce user management per console

### LINKED ISSUE #8245: Feature Request: add user management to bin/console
Hi!
Currently there is no way to manage users in bin/console(.*), like add, delete, edit, search... There is only an option to set a new password. 

It would greatly helpful when there would be such an option so that you can manage users the same way like you can in web under /admin/users.

### PR DESCRIPTION
**Pull request by nupplaphil**
Closes #8245 
Part of #4090 

Possible commands:
- `bin/console user password <nickname> [<password>] [-h|--help|-?] [-v]`
- `bin/console user add [<name> [<nickname> [<email> [<language>]]]] [-h|--help|-?] [-v]`
- `bin/console user delete [<nickname>] [-q] [-h|--help|-?] [-v]`
- `bin/console user allow [<nickname>] [-h|--help|-?] [-v]`
- `bin/console user deny [<nickname>] [-h|--help|-?] [-v]`
- `bin/console user block [<nickname>] [-h|--help|-?] [-v]`
- `bin/console user unblock [<nickname>] [-h|--help|-?] [-v]`
- `bin/console user list pending [start=0 [count=50]] [-h|--help|-?] [-v]`
- `bin/console user list removed [start=0 [count=50]] [-h|--help|-?] [-v]`
- `bin/console user list all [start=0 [count=50]] [-h|--help|-?] [-v]`
- `bin/console user search id <UID> [-h|--help|-?] [-v]`
- `bin/console user search nick <nick> [-h|--help|-?] [-v]`
- `bin/console user search mail <mail> [-h|--help|-?] [-v]`
- `bin/console user search guid <GUID> [-h|--help|-?] [-v]`

ToDos:
- <strike>Add resend pending mail/registration mail</strike> maybe in a follow up :)
- [x] Add search/list functionality (But just basic possibilities)
- [x] Full test of all console commands
- [x] Full test of all admin GUI commands

### CODE REVIEW

File: src/Console/User.php [hunk 1]
```diff
@@ -63,6 +64,13 @@ protected function getHelp()
 	bin/console user deny [<nickname>] [-h|--help|-?] [-v]
 	bin/console user block [<nickname>] [-h|--help|-?] [-v]
 	bin/console user unblock [<nickname>] [-h|--help|-?] [-v]
+	bin/console user list pending [start=0 [count=50]] [-h|--help|-?] [-v]
```
Comment (MrPetovan) [inline src/Console/User.php]: Shouldn't start and count be named parameters like `--start` or `-s` and `--count` or `-n` instead of positional arguments?

Comment (nupplaphil) [reply src/Console/User.php]: mhmm .. good idea

**Comment by MrPetovan** [comment]
> @nupplaphil added 3 commits in 7 minutes

Whoa, you're committing in the future!

**Comment by nupplaphil** [comment]
> > @nupplaphil added 3 commits in 7 minutes
> 
> Whoa, you're committing in the future!

Ah damn .. I now see that my ubuntu is switching again 1 hour into the future -.-** The NTP service is stopping sometimes and my hardware clock seems broken/not right ..

**Comment by MrPetovan** [comment]
is this done?

**Comment by nupplaphil** [comment]
Yep, I'm not adding any new functions(yet).

Maybe the node admins will have some ideas which user management commands will be useful for them :) .

File: src/Model/User.php [hunk 2]
```diff
@@ -1323,21 +1323,38 @@ public static function getStatistics()
...
 	 * @param int    $count Count of the items per page (Default is @see Pager::ITEMS_PER_PAGE)
+	 * @param string $type  The type of users, which should get (all, bocked, removed)
 	 * @param string $order Order of the user list (Default is 'contact.name')
 	 * @param string $order_direction Order direction (Default is ASC)
 	 *
 	 * @return array The list of the users
 	 * @throws Exception
 	 */
-	public static function getUsers($start = 0, $count = Pager::ITEMS_PER_PAGE, $order = 'contact.name', $order_direction = '+')
+	public static function getUsers($start = 0, $count = Pager::ITEMS_PER_PAGE, $type = 'all', $order = 'contact.name', $order_direction = '+')
```
Comment (MrPetovan) [inline src/Model/User.php]: Please rename to `getList`, it's already the User model, no need to repeat it.

**Comment by MrPetovan** [comment]
Test is failing generally.

**Comment by nupplaphil** [comment]
I'll rebase it after #8348  :)

**Comment by MrPetovan** [comment]
I don't think you need to unless there's a merge conflict.

**Comment by nupplaphil** [comment]
too late ^^..

**Comment by nupplaphil** [comment]
you're right .. It would have been much easier to just restart the test at travis instead ..

**Pull request merged**